mycroft/skills/core.py

```diff
@@ -452,8 +454,8 @@ def show_url(self, url, override_idle=None):
 
     def write_html_to_temp(self, html):
         """ Writes html from a skill to temp file in system /tmp location """
-        fname = '/tmp/tempfile.html'
-        f = open(fname,'w')
+        fname = "/tmp/tempfile.html"
+        f = open(fname, 'w')
```

Comment:
Wondering if the [loadHtml](https://doc.qt.io/qt-5/qml-qtwebengine-webengineview.html#loadHtml-method) could be used instead of saving to a temp file?

Might be better when not sharing the filesystem

Reply:
Have re-implemented it using webengineview loadHtml(), this removes filesystem storage of html itself but webengineview.loadHtml() requires a resource parameter to load resources (images, scripts, etc) from the correct location. This parameter can be a local or remote url, if None reverts to "http://localhost".

```python
raw_html = """ <img src="apple.png" width=100px height=100px"> """
self.html_resource = 'file://' + dirname(__file__) + '/res/' #resource in skill folder
self.gui.show_html(raw_html, resource_url=self.html_resources, override_idle=True)
```